legend-engine-xt-serviceStore-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/ServiceStoreParseTreeWalker.java

```diff
@@ -368,16 +427,18 @@ private TypeReference visitTypeReference(ServiceStoreParserGrammar.TypeReference
         return typeReference;
     }
 
-    private SecurityScheme visitSecurityScheme(ServiceStoreParserGrammar.IdentifierContext securitySchemeCtx)
+    private SecuritySchemeRequirement visitSecurity(ServiceStoreParserGrammar.IdentifierContext securityCtx)
     {
-        String securityScheme = PureGrammarParserUtility.fromIdentifier(securitySchemeCtx);
-        List<Function<String, SecurityScheme>> processors = ListIterate.flatCollect(IServiceStoreGrammarParserExtension.getExtensions(), ext -> ext.getExtraSecuritySchemesParsers());
+        String securitySchemeId = securityCtx.getText();
```

Comment:
why can't this be simple map lookup?

Reply:
This is to accommodate old grammar

Reply:
Lets follow this decision tree
- Simple map lookup, if we find something return
- Use the existing extension to see if for identifier we can get some legacy SecurityScheme, if yes add to map and return
- Throw error